src/dune_engine/fs_memo.ml

```diff
@@ -71,24 +108,24 @@ let invalidate_path_and_its_parent path =
     | Some path -> invalidate_path path)
 
 module Event = struct
-  (* Here are some assumptions about events:
+  (* Here are some idealized assumptions about events:
+
+     - If a file is renamed, we receive [Created] and [Deleted] events with
+     corresponding paths.
 
-     - If a file is renamed, we receive [File_created] and [File_deleted] events
-     with corresponding paths.
+     - If a directory is renamed then in addition to the [Created] and [Deleted]
+     events for the directory itself, we receive events about all file and
+     directory paths in the corresponding file tree.
 
-     - If a directory is renamed then in addition to the [Directory_created] and
-     [Directory_deleted] events for the directory itself, we receive events
-     about all file and directory paths in the corresponding file tree.
+     - Similarly, if a directory is deleted, we receive the [Deleted] event for
+     the directory itself, as well as deletion events for all watched paths in
+     the corresponding file tree.
 
-     - Similarly, if a directory is deleted, we receive the [Directory_deleted]
-     event for the directory itself, as well as deletion events for all paths in
-     the corresponding file tree. *)
+     Very little of these assumptions currently hold. *)
```

Comment:
If you have some bits of knowledge about which assumptions currently hold and which don't, it seems worth writing that down.

Reply:
Changed to:

```
     Not all of these assumptions currently hold. In particular, directory
     renames probably just give "created" and "deleted" for the directory
     itself, which means we are not correctly handling directory renames. 
```